File: utils/institutional_email.py
import os


def _allowed_domains() -> frozenset[str]:
    raw = os.getenv("LMCP_INSTITUTIONAL_EMAIL_DOMAINS", "lmcp.ac.in")
    return frozenset(part.strip().lower() for part in raw.split(",") if part.strip())


def normalize_email(email: str) -> str:
    return email.strip().lower()


def email_domain(email: str) -> str:
    normalized = normalize_email(email)
    if "@" not in normalized:
        return ""
    return normalized.rsplit("@", 1)[-1]


def is_lmcp_institutional_email(email: str) -> bool:
    domain = email_domain(email)
    if not domain:
        return False
    return domain in _allowed_domains()

File: frontend/src/utils/institutionalEmail.ts
const defaultDomains = "lmcp.ac.in";

export function getAllowedInstitutionalDomains(): string[] {
  const raw =
    import.meta.env.VITE_LMCP_INSTITUTIONAL_EMAIL_DOMAINS ?? defaultDomains;
  return raw
    .split(",")
    .map((domain) => domain.trim().toLowerCase())
    .filter(Boolean);
}

export function isLmcpInstitutionalEmail(email: string): boolean {
  const normalized = email.trim().toLowerCase();
  const atIndex = normalized.lastIndexOf("@");
  if (atIndex < 0) {
    return false;
  }
  const domain = normalized.slice(atIndex + 1);
  return getAllowedInstitutionalDomains().includes(domain);
}

export function institutionalEmailHint(): string {
  const domains = getAllowedInstitutionalDomains();
  if (domains.length === 1) {
    return `@${domains[0]}`;
  }
  return domains.map((domain) => `@${domain}`).join(", ");
}
